one-ui/src/main/java/com/marklogic/hub/oneui/auth/LoginLogoutHandler.java

```diff
@@ -43,19 +42,35 @@ public void onAuthenticationSuccess(HttpServletRequest request, HttpServletRespo
         ConnectionAuthenticationToken authenticationToken = (ConnectionAuthenticationToken)authentication;
         ObjectNode resp = mapper.createObjectNode();
         boolean isInstalled = authenticationToken.stagingIsAccessible();
+        final boolean[] managePrivilege = new boolean[3];
         if (authenticationToken.stagingIsAccessible()) {
             try {
                 RolesService rolesService = RolesService.on(authenticationToken.getHubConfigSession().newStagingClient());
                 resp.putArray("roles").addAll((ArrayNode) rolesService.getRoles());
+                if (authenticationToken.hasManagePrivileges()) {
+                    ArrayNode rolesNodes = (ArrayNode) resp.get("roles");
+                    rolesNodes.forEach(e -> {
+                        if ("data-hub-environment-manager".equals(e.asText())) {
+                            managePrivilege[0] = true;
+                            return;
+                        } else if ("manager-admin".equals(e.asText())) {
+                            managePrivilege[1] = true;
+                        } else if ("security".equals(e.asText())) {
+                            managePrivilege[2] = true;
+                        }
+                    });
+                }
             } catch (FailedRequestException e) {
                 // If Roles Data Service isn't installed, the latest Data Hub isn't installed
                 if (e.getServerStatusCode() == HttpStatus.SC_NOT_FOUND) {
                     isInstalled = false;
                 }
             }
         }
+
         resp.put("isInstalled", isInstalled);
-        resp.put("hasManagePrivileges", authenticationToken.hasManagePrivileges());
+        resp.put("hasManagePrivileges", !authenticationToken.hasManagePrivileges() ? false :
+            (managePrivilege[0] || managePrivilege[1] && managePrivilege[2]) ? true : false);
```

Comment:
We should consider renaming hasManagePrivileges to something else. It seems like we are comparing privilege to a role.
Can we simplify this, by having a variable for line 73. And arr[0]/[1]/[2] dont read too well.. maybe something like envManager || manageAdmin && security

Reply:
added test case. It is a temporary workaround and we will revisit it and once a better function exposed from RoleService. And right now one array defintion is better than 3 arrays for lamda.